Automattic/woocommerce-payments
# Hide UPE settings behind flag. Update settings page design

Fixes #1914

#### Changes proposed in this Pull Request





* To be able to control visibility of the "Block appearance" section based on UPE feature flag, extract this section to a separate `BlockAppearance` React component.
    * Extract its sibling, `DebugMode`, for consistency.
    * Extract `CheckboxToggle` and `useToggle`.
* Introduce `WCPaySettingsContext` to pass `window.wcpaySettings` to React elements.
    * Use this to pass feature flag to `AdvancedSettings` to decide whether to display `BlockAppearance`.
    * Rewrite existing code that uses `accountLink` passed as prop to also use context instead.
* Reorganize cards and update copy as per the post mentioned in #1914.
* Move existing tests to reflect current component organization.
* Add `WC_Payments_Features::to_array()` method to generate array of feature flags for output on the front-end.





#### Testing instructions





1. Verify that when `_wcpay_feature_upe` is not `'1'`, the "Block appearance" section is not displayed in Advanced settings.
2. Open the Figma file linked to in the P2 post from #1914 and verify the changes match the design.

#### Questions

1. Should all available feature flags always be exposed to the front-end or just the enabled ones? At the moment the code exposes all supported flags (in line with the previous behavior for the grouped settings flag) in the `WC_Payments_Features::to_array()` method.
2. I did *not* remove the Payments accepted on checkout card as it's hidden if there is just one payment method available anyway but if I should do it, please let me know!

-------------------

    - Not added - feature behind feature flag.
